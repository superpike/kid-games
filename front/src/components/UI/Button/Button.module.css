.Button {
    background-color: transparent;
    border: none;
    color: white;
    outline: none;
    cursor: pointer;
    font: inherit;
    display: block;
    padding: 10px;
    line-height: 1;
    border-radius: 300px;
    box-shadow: 0 1px 3px hsla(0, 0%, 0%, .2);
}

.Middle {
    margin: 0 auto;
}

.Main {
    background-color: #009b00;
    min-width: 200px;
    margin: 20px;
}

.Main:hover {
    transform: translate(0,-1px);
    background-color: #00d300;
}

.Main:active {
    transform: translate(0,0);
}

.Danger {
    background-color: #7a1111;
}

.Danger:hover {
    background-color: #ff0000;
}

.Auth {
    background-color: #1e11dd;
}

.Auth:hover {
    background-color: #1100fd;
}

.Button:disabled {
    background-color: #D9E2EC;
    color: #9FB3C8;
    cursor: not-allowed;
}