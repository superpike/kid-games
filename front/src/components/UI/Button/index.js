import React from 'react';
import PropTypes from 'prop-types';

import classes from './Button.module.css';

const Button = ({ btnType, disabled, clicked, children, testId }) => {
  let assignedClasses = [];
  if (btnType) {
    assignedClasses = btnType.split(' ');
  }
  assignedClasses.push('Button');

  return (
    <button
      disabled={disabled}
      className={assignedClasses
        .map(el => {
          return classes[el];
        })
        .join(' ')}
      onClick={clicked}
      type="button"
      data-testid={testId}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  btnType: PropTypes.oneOf([
    'Main Middle',
    'Middle',
    'Main',
    'Auth',
    'Danger',
    'Auth Middle',
    'Danger Middle',
  ]).isRequired,
  disabled: PropTypes.bool,
  clicked: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired,
  testId: PropTypes.string,
};

Button.defaultProps = {
  disabled: false,
  testId: '',
};

export default Button;
