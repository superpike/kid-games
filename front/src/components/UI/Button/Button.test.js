import React from 'react';
import Button from './index';
// query utilities:
// import {
//     getByLabelText,
//     getByText,
//     getByTestId,
//     getByDisplayValue,
//     // Tip: all queries are also exposed on an object
//     // called "queries" which you could import here as well
//     wait,
// } from '@testing-library/dom'
// adds special assertions like toHaveTextContent
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'

describe("Testing button", () => {
    // describe("classes", () => {
        test('should be only Main', () => {
            const { container } = render(
                <Button btnType='Main' clicked={() => {}}>Button</Button>
            );
            expect(container).toMatchSnapshot();
        })
        test('should be only Middle', () => {
            const { container } = render(
                <Button btnType='Middle' clicked={() => {}}>Button</Button>
            );
            expect(container).toMatchSnapshot();
        })
        test('should be all classes', () => {
            const { container } = render(
                <Button btnType='Main Middle' clicked={() => {}}>Button</Button>
            );
            expect(container).toMatchSnapshot();
        })
    // })
})