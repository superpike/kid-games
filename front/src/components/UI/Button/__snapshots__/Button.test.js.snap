// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button should be all classes 1`] = `
<div>
  <button
    class="Main Middle Button"
    data-testid=""
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`Testing button should be only Main 1`] = `
<div>
  <button
    class="Main Button"
    data-testid=""
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`Testing button should be only Middle 1`] = `
<div>
  <button
    class="Middle Button"
    data-testid=""
    type="button"
  >
    Button
  </button>
</div>
`;
